src/processor/queue-consumer.service.ts

```diff
@@ -148,24 +167,28 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
   private async checkCapacity(): Promise<void> {
     const capacityLimit = this.configService.getCapacityLimit();
     const capacity = await this.blockchainService.capacityInfo(this.configService.getProviderId());
-    let outOfCapacity = false;
-
-    if (capacityLimit.type === 'percentage') {
-      const minRemainingPct = 100 - capacityLimit.value;
-      const percentageRemaining = (capacity.remainingCapacity * 100n) / (capacity.totalCapacityIssued || 1n);
-      if (percentageRemaining <= minRemainingPct) {
-        outOfCapacity = true;
-        this.logger.warn(
-          `Capacity threshold reached: remaining pct = ${percentageRemaining}% (${minRemainingPct}% required (${
-            (capacity.remainingCapacity * 100n) / capacity.totalCapacityIssued
-          }))`,
-        );
-      }
-    } else {
-      const capacityUsed = capacity.totalCapacityIssued - capacity.remainingCapacity;
-      if (capacityUsed >= capacityLimit.value) {
-        outOfCapacity = true;
-        this.logger.warn(`Capacity threshold reached: used ${capacityUsed} of ${capacityLimit.value}`);
+    const remainingCapacity = capacity.remainingCapacity;
+    let outOfCapacity = capacity.remainingCapacity <= 0n;
```

Comment:
if provider hasn't stake yet, it would be quick to check that and set `outOfCapacity` as true